Mathlib/CategoryTheory/Monoidal/Braided.lean

```diff
@@ -618,4 +651,100 @@ attribute [reassoc] associator_monoidal
 
 end Tensor
 
+namespace MonoidalOpposite
+
+instance instBraiding : BraidedCategory Cᴹᵒᵖ where
+  braiding X Y := (β_ (unmop Y) (unmop X)).mop
+  braiding_naturality_right X {_ _} f := braiding_naturality_left f.unmop (unmop X)
+  braiding_naturality_left {_ _} f Z := braiding_naturality_right (unmop Z) f.unmop
+  hexagon_forward X Y Z := hexagon_reverse (unmop Z) (unmop Y) (unmop X)
+  hexagon_reverse X Y Z := hexagon_forward (unmop Z) (unmop Y) (unmop X)
+
+/-- The identity functor on `C`, viewed as a functor from `C` to its
+monoidal opposite, upgraded to a braided functor. -/
+@[simps!] def mopBraidedFunctor : BraidedFunctor C Cᴹᵒᵖ where
+  μ X Y := (β_ (mop X) (mop Y)).hom
+  ε := 𝟙 (𝟙_ Cᴹᵒᵖ)
+  associativity X Y Z := by
+    simp only [tensorHom_id, id_tensorHom]
```

Comment:
I think we need just `simp [← yang_baxter_assoc]` here. We may miss some simp lemmas like `mop (α_ X Y Z).hom = (α_ (mop Z) (mop Y) (mop X)).inv`.

Reply:
I got (roughly) this to work by adding `@[simps]` to `monoidalCategoryMop`

Reply:
For `unmopBraidedFunctor` I couldn't get it to happen just with simp, although the proof is much cleaner now. Also it seems that that `@[simps]` attribute breaks the ability of `aesop_cat` to prove `μ_natural_left, μ_natural_right` for `mopBraidedFunctor`. Do you have any suggestions?

Reply:
Ah, I see why you suggested `mop (α_ X Y Z).hom = (α_ (mop Z) (mop Y) (mop X)).inv` as a simp lemma. That does seem much nicer. I've started adding lemmas like this for mop/op but simp now runs into a stack overflow, so I need to debug first

Reply:
> Also it seems that that @[simps] attribute breaks the ability of aesop_cat to prove μ_natural_left, μ_natural_right for mopBraidedFunctor.

I think this suggests that adding @[simps] is not a good idea for these instances. In fact, the lemmas provided by simps (which can be seen by putting `set_option trace.simps.verbose true in`)  are different from what we want. I think the following lemmas are helpful, but unfortunately we should add them manually:
```lean
@[simp] theorem mop_associator_hom (X Y Z : C) : (α_ X Y Z).hom.mop = (α_ (mop Z) (mop Y) (mop X)).inv := rfl
@[simp] theorem unmop_associator_hom (X Y Z : Cᴹᵒᵖ) : (α_ X Y Z).hom.unmop = (α_ (unmop Z) (unmop Y) (unmop X)).inv := rfl
```

Reply:
Yes, I definitely agree putting `@[simps]` where I did is a mistake. I'm in the process of adding lemmas like what you've written and making the existing code use them

Reply:
The code has been updated with these lemmas